### PR TITLE
Add mark-as-read, mark-as-unread, clear-all-completed-todo featues

### PR DESCRIPTION
**Pull request by Aqurds**
The following features were added:

- Mark todo as read
- Mark todo as unread
- Delete all completed todo
- Data stored and updated in localStorage

### CODE REVIEW

File: src/modules/localStorage.js [hunk 1]
```diff
@@ -107,4 +107,77 @@ export default class toDoStorage {
...
+      JSON.stringify(currentTodoData)
+    );
+  }
+
+  clearAllCompletedTodo() {
+    // Responsible to clear all completed todo, update data storage
+    const currentTodoData = JSON.parse(
+      localStorage.getItem(this.localStorageObjectName)
+    );
+
```
Comment (souad988) [inline src/modules/localStorage.js:135]: You have done great by implementing all functionalities 👍🏼  , I would just suggest that you separate get and set localStorage from your apps main functionalities for a DRY and better performant code call `localStorage.getItem(this.localStorageObjectName)` only once onload and save the content in a local object then whenever you want to perform an edit or delete action update your local object then set localStorage item with the new updated object

File: src/modules/localStorage.js [hunk 2]
```diff
@@ -107,4 +107,77 @@ export default class toDoStorage {
...
+      for (let i = 0; i < currentTodoData.length; i += 1) {
+        // console.log(currentTodoData[i]);
+        if (currentTodoData[i].completed !== true) {
+          newTodo.push({
+            index: newIndex,
+            task: currentTodoData[i].task,
+            completed: currentTodoData[i].completed,
+          });
+          newIndex += 1;
+        }
```
Comment (souad988) [inline src/modules/localStorage.js]: Great your approach is working fine but in this projects requirements you were asked to use array.filter method

File: src/modules/localStorage.js [hunk 3]
```diff
@@ -107,4 +107,77 @@ export default class toDoStorage {
...
+        // console.log(element);
+        element.completed = false;
+      }
+    });
+
+    localStorage.setItem(
+      this.localStorageObjectName,
+      JSON.stringify(currentTodoData)
+    );
+  }
```
Comment (souad988) [inline src/modules/localStorage.js]: after creating a local object you wont need to call localStorage getItem and these two functions can be merged into only one that you can call toggleMark or toggleComplete `element.completed = !element.completed`

**Review by souad988 — CHANGES_REQUESTED** [review]
Hi @Aqurds  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
Add mark as complete functionality
Add clearallCompleted
Create js module

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/localStorage.js (hunk 3 above)
Comment (Aqurds) [reply src/modules/localStorage.js:147]: @souad988 I prefer this coding style. Why you are forcing me to follow your style? Is it required for project approval? A reviewer is meant to find out the following key requirements and bugs. Not to follow any specific coding style.

File: src/modules/localStorage.js (hunk 2 above)
Comment (Aqurds) [reply src/modules/localStorage.js]: @souad988 I will follow this suggestion. Thanks

File: src/modules/localStorage.js (hunk 1 above)
Comment (Aqurds) [reply src/modules/localStorage.js:154]: @souad988 same things here. You are making your suggestion as a requirement for project approval. I don't like it!

File: src/modules/localStorage.js (hunk 2 above)
Comment (Aqurds) [reply src/modules/localStorage.js]: array.filter method added

File: src/modules/localStorage.js (hunk 3 above)
Comment (wayungi) [reply src/modules/localStorage.js]: - [x] A code reviewer's role is to guide you to write better code. Code that has a lot of duplication is hard to maintain. It is therefore my humble suggestion that you write DRY code. Please update you code inline with[ this suggestion](https://github.com/Aqurds/t-d-l/pull/3/files#r1232164472) 👍🏾

**Review by wayungi — CHANGES_REQUESTED** [review]
Hi @Aqurds , 

image

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.


 ## Highlights
- [x] Good commit messages ✔
- [x]  Correct GitHub flow ✔️ 
- [x] Descriptive PR body 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/localStorage.js (hunk 3 above)
Comment (Aqurds) [reply src/modules/localStorage.js]: Applied this suggestion. Thanks!

File: src/modules/localStorage.js (hunk 1 above)
Comment (Aqurds) [reply src/modules/localStorage.js:135]: Need more clarification on this. The data in local storage is constantly updating, so I think this approach will not work ( I tried, but didn't work). If you have any specific coding pattern in your mind, we may discuss that in a video call. Thanks.

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Aqurds, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




### To Highlight :1st_place_medal: 

- Descriptive Pull Request title & summary. ✔️  
- Add and Remove functionality works Perfectly ✔️ 
- Edit Task functionality works fine ✔️ 
- Good use of Github flow ✔️ 

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**